internal/services/mobilenetwork/mobile_network_attached_data_network_resource.go

```diff
@@ -0,0 +1,617 @@
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			attachedDataNetwork := &attacheddatanetwork.AttachedDataNetwork{
+				Location: location.Normalize(model.Location),
+				Properties: attacheddatanetwork.AttachedDataNetworkPropertiesFormat{
+					DnsAddresses:                         model.DnsAddresses,
+					UserEquipmentAddressPoolPrefix:       &model.UserEquipmentAddressPoolPrefix,
+					UserEquipmentStaticAddressPoolPrefix: &model.UserEquipmentStaticAddressPoolPrefix,
+					UserPlaneDataInterface:               attacheddatanetwork.InterfaceProperties{},
+				},
+				Tags: &model.Tags,
+			}
+
+			attachedDataNetwork.Properties.NaptConfiguration = expandNaptConfigurationModel(model.NaptConfiguration)
+
+			if model.UserPlaneAccessName != "" {
+				attachedDataNetwork.Properties.UserPlaneDataInterface.Name = &model.UserPlaneAccessName
+			}
+
+			if model.UserPlaneAccessIPv4Address != "" {
+				attachedDataNetwork.Properties.UserPlaneDataInterface.IPv4Address = &model.UserPlaneAccessIPv4Address
+			}
+
+			if model.UserPlaneAccessIPv4Subnet != "" {
+				attachedDataNetwork.Properties.UserPlaneDataInterface.IPv4Subnet = &model.UserPlaneAccessIPv4Subnet
+			}
+
+			if model.UserPlaneAccessIPv4Gateway != "" {
+				attachedDataNetwork.Properties.UserPlaneDataInterface.IPv4Gateway = &model.UserPlaneAccessIPv4Gateway
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, *attachedDataNetwork); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r AttachedDataNetworkResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 180 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MobileNetwork.AttachedDataNetworkClient
+
+			id, err := attacheddatanetwork.ParseAttachedDataNetworkID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var plan AttachedDataNetworkModel
+			if err := metadata.Decode(&plan); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			addtachedDataNetwork := resp.Model
+			if addtachedDataNetwork == nil {
+				return fmt.Errorf("retrieving %s: addtachedDataNetwork was nil", id)
+			}
+
+			if metadata.ResourceData.HasChange("dns_addresses") {
+				addtachedDataNetwork.Properties.DnsAddresses = plan.DnsAddresses
```

Comment:
Since this (and the following properties) are optional, which in the `Create` function, you only set them when they are "non-zero", then shall we also do the empty check here to nil set them when the new value is of zero value?